querydsl-jpa/src/main/java/com/querydsl/jpa/HibernateHandler.java

```diff
@@ -83,9 +78,10 @@ public boolean createNativeQueryTyped() {
 
     @Override
     public boolean transform(Query query, FactoryExpression<?> projection) {
-        if (query instanceof HibernateQuery) {
+        if (query instanceof NativeQuery) {
             ResultTransformer transformer = new FactoryExpressionTransformer(projection);
-            ((HibernateQuery) query).getHibernateQuery().setResultTransformer(transformer);
+            //noinspection deprecation
```

Comment:
Remove comment? Reading/reviewing the code it means nothing to me.

Reply:
Thats suppresing Intellij warnings if I'm right

Reply:
> That's suppressing Intellij warnings if I'm right

Yep. setResultTransformation was marked Deprecated with no alternative but to use it and suppress warning until Hibernate 6 comes out.

Reply:
Cool

Reply:
having that in mind you could annotate the method with `@SuppressWarnings("deprecation")` as that will suppress both IDE and compiler